include/z64player.h

```diff
@@ -655,7 +655,7 @@ typedef struct WeaponInfo {
 #define PLAYER_STATE1_HOSTILE_LOCK_ON (1 << 4) // Currently locked onto a hostile actor. Triggers a "battle" variant of many actions.
 #define PLAYER_STATE1_5 (1 << 5)
 #define PLAYER_STATE1_6 (1 << 6)
-#define PLAYER_STATE1_7 (1 << 7)
+#define PLAYER_STATE1_DEAD (1 << 7) // Player has died. Note that this gets set when the death cutscene has started, after landing from the air.
```

Comment:
i'm a little iffy on this name, I actually prefer `PLAYER_STATE1_IN_DEATH_CUTSCENE`. not sure if you will agree with my reasons but I will try to justify:

- `func_80843AE8` (the "fairy revival" process) sets the flag once Link is back to full health, signaling the end of the "cutscene" and returning to his normal Idle actionFunc. during the whole process, even when Link has HP, the flag remains. having HP and being "STATE1_DEAD" feels incompatible
- in `Player_UpdateCommon`, the check that starts the death process by actually checking `gSaveContext.save.info.playerData.health`, if the player is climbing it will first end the climb and then start the death process. this further makes me see it as a flag tied to the death cutscene.
- It freezes other actors, like what happens in other cutscenes e.g. bottles, din's etc
- it's checked as part of `Player_InBlockingCsMode` which is checking all kinds of other cutscene state
- this is probably the least compelling reason but I thought I'd include it anyway: but if you just set this flag forcibly, Link doesn't actually die, you still have control and things just behave oddly. the only way to make Link "dead" is setting health to 0

Reply:
> Confused on point 1, it UNSETS it when hes healed, because he is no longer dead.

woops mb yea I meant "unsets". the intent was to say he is in a state where he currently has HP but the flag is still set, and is only cleared once back to full hp.

Ok, I don't feel strongly enough to argue for it further, I definitely don't think it's bad or wrong or anything. if you still prefer `DEAD` then LGTM 👍

Reply:
Its also possible to reach 0 health in the air and land on a heart, in which case you never actually die. its one of the more obscure ways to achieve the "infinite sword glitch" as well lol

Anyway yeah open to hearing more thoughts on this. It does kind of boil down to a philosophical question of "what does it mean for Link to be dead"